.codegen/service.go.tmpl

```diff
@@ -182,22 +184,18 @@ func new{{.PascalName}}() *cobra.Command {
 					return fmt.Errorf("expected to have {{range .Request.RequiredFields}}{{.Summary | trimSuffix "." | lower}}{{end}}")
 				}
 			{{- end -}}
+
 			{{$method := .}}
-			{{- if and .Request.IsAllRequiredFieldsPrimitive (not .IsJsonOnly) -}}
-				{{- range $arg, $field := .Request.RequiredFields}}
-					{{if not $field.Entity.IsString -}}
-					_, err = fmt.Sscan(args[{{$arg}}], &{{$method.CamelName}}Req.{{$field.PascalName}})
-					if err != nil {
-						return fmt.Errorf("invalid {{$field.ConstantName}}: %s", args[{{$arg}}])
-					}{{else -}}
-					{{$method.CamelName}}Req.{{$field.PascalName}} = args[{{$arg}}]
-					{{- end -}}{{end}}
-			{{- else -}}
-				return fmt.Errorf("please provide command input in JSON format by specifying the --json flag")
+			{{- range $arg, $field := .Request.RequiredFields}}
```

Comment:
If the JSON flag is specified, this loop should only happen for the path parameters.

Alternatively, we treat the args as optional and we overwrite the parameters passed in the JSON body.

Either way, we can't run this loop in all cases, because `args` may not have enough entries.